specifications/SimplifiedFastPaxos/Paxos.tla

```diff
@@ -0,0 +1,208 @@
+
+    Else, it can ignore the Accept message or request.
+*)
+PaxosAccepted ==
+    /\ UNCHANGED<<decision>>
+    /\ \E a \in Replicas, m \in p2aMessages :
+        /\ m.value \in Values
+        /\ maxBallot[a] <= m.ballot
+        /\ maxBallot' = [maxBallot EXCEPT ![a] = m.ballot]
+        /\ maxVBallot' = [maxVBallot EXCEPT ![a] = m.ballot]
+        /\ maxValue' = [maxValue EXCEPT ![a] = m.value]
+        /\ SendMessage([type |-> "P2b",
+                        ballot |-> m.ballot,
+                        acceptor |-> a,
+                        value |-> m.value])
+
+(*
+    Consensus is achieved when a majority of acceptors accept the same ballot number (rather than the same value).
+    Because each ballot is unique to a proposer and only one value may be proposed per ballot, all acceptors
+    that accept the same ballot thereby accept the same value.
+
+    There is no need to model the variable decision for every acceptor. In this specification, the variable decision
+    represents the decision of any acceptor, can its value may potentially be changed multiple times. Instead, we use
+    the consistency safety property to proof that the decision for every acceptor is the same.
+*)
+PaxosDecide ==
+    /\ UNCHANGED<<messages, maxBallot, maxVBallot, maxValue>>
+    /\ \E b \in Ballots, q \in Quorums :
+        LET M == {m \in p2bMessages : m.ballot = b /\ m.acceptor \in q}
+        IN /\ \A a \in q : \E m \in M : m.acceptor = a
+           /\ \E m \in M : decision' = m.value
+
+PaxosTypeOK == /\ messages \subseteq Message
+               /\ decision \in Values \union {none}
+               /\ maxBallot \in [Replicas -> Ballots]
+               /\ maxVBallot \in [Replicas -> Ballots]
+               /\ maxValue \in [Replicas -> Values \union {none}]
+
+PaxosInit == /\ messages = {}
+             /\ decision = none
+             /\ maxBallot = [r \in Replicas |-> 0]
+             /\ maxVBallot = [r \in Replicas |-> 0]
+             /\ maxValue = [r \in Replicas |-> none]
+
+PaxosNext == \/ PaxosPrepare
+             \/ PaxosPromise
+             \/ PaxosAccept
+             \/ PaxosAccepted
+             \/ PaxosDecide
+
+PaxosSpec == /\ PaxosInit
+             /\ [][PaxosNext]_<<messages, decision, maxBallot, maxVBallot, maxValue>>
+             /\ SF_<<messages, decision, maxBallot, maxVBallot, maxValue>>(PaxosDecide)
+
+\* Only proposed values can be learnt.
+PaxosNontriviality ==
+    /\ \/ decision = none
+       \/ \E m \in p2aMessages : m.value = decision
+    /\ \A m \in p1bMessages : /\ m.maxValue \in Values \/ 0 = m.maxVBallot
+                              /\ m.maxValue = none \/ 0 < m.maxVBallot
+
+\* At most 1 value can be learnt.
+PaxosConsistency == decision = none \/ decision = decision'
+
+PaxosSafetyProperty == /\ [][PaxosNontriviality]_<<messages, decision, maxBallot, maxVBallot, maxValue>>
+                       /\ [][PaxosConsistency]_<<messages, decision, maxBallot, maxVBallot, maxValue>>
```

Comment:
Since PaxosNontriviality is a state predicate, it can be checked as an invariant rather than as a temporal property.
Also, the second conjunct can be written a little more concisely as [][decision = none]_<<decision>>, which asserts that a decision can be taken only if its current value is none.

Reply:
Ah, Github pretty-printing ... That formula was
```tla
[][decision = none]_<< decision >>
```